src/Sonarr.Api.V5/RootFolders/RootFolderController.cs

```diff
@@ -0,0 +1,73 @@
+using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using NzbDrone.Core.RootFolders;
+using NzbDrone.Core.Validation.Paths;
+using NzbDrone.SignalR;
+using Sonarr.Http;
+using Sonarr.Http.Extensions;
+using Sonarr.Http.REST;
+using Sonarr.Http.REST.Attributes;
+
+namespace Sonarr.Api.V5.RootFolders;
+
+[V5ApiController]
+public class RootFolderController : RestControllerWithSignalR<RootFolderResource, RootFolder>
```

Comment:
A rare opportunity to fix the endpoint name from singular to plural? `/rootfolders{/$id}` would more appropriate than `/rootfolder{/$id}`.

```suggestion
public class RootFoldersController : RestControllerWithSignalR<RootFolderResource, RootFolder>
```

Reply:
I'm not sure plural or singular really matters, so long as we're consistent, though I'm not opposed to it, but I think it warrants a bit larger discussion and if we decide to, do it as a change once everything has been migrated before shipping v5 to make sure we get everything.